docs/Releases.md

```diff
@@ -13,6 +13,307 @@ use a date based versioning system.  Now, a release version can look
 like `17.11` where the "major" number is the year and the "minor" number
 is the month.
 
+## v19.05 (5/19): Shared Core Mode, Major Architectural/API/Initialization/Signal Handling Changes, Advanced Rings Changes, Stats Updates, CI PR Review, LPM Firewall NF, Payload Search NF, TTL Flags, minor improvements and bug fixes.
+This release adds several new features and changes how the onvm_mgr and NFs start. A CloudLab template is available with the latest release here: https://www.cloudlab.us/p/GWCloudLab/onvm
+
+**This release features a lot of breaking API changes.**
+
+**Performance**: This release should fix the major performance issues that were present in the last release. 
+
+**Repo changes**: Default branch has been changed to `master`, active development can still be seen in `develop`. Most of the development is now done on the public repo to improve visibility, planned projects and improvements can be seen in this [pinned issue](https://github.com/sdnfv/openNetVM/issues/91), additionally pull requests and issues are now cataloged by tags.
+
+### Shared Core Mode:
+This code introduces **EXPERIMENTAL** support to allow NFs to efficiently run on **shared** CPU cores. NFs wait on semaphores when idle and are signaled by the manager when new packets arrive. Once the NF is in wake state, no additional notifications will be sent until it goes back to sleep. Shared core variables for mgr are in the `nf_wakeup_info` structs, the NF shared core vars were moved to the `onvm_nf` struct.
+
+The code is based on the hybrid-polling model proposed in [_Flurries: Countless Fine-Grained NFs for Flexible Per-Flow Customization_ by Wei Zhang, Jinho Hwang, Shriram Rajagopalan, K. K. Ramakrishnan, and Timothy Wood, published at _Co-NEXT 16_][flurries_paper] and extended in [_NFVnice: Dynamic Backpressure and Scheduling for NFV Service Chains_ by Sameer G. Kulkarni, Wei Zhang, Jinho Hwang, Shriram Rajagopalan, K. K. Ramakrishnan, Timothy Wood, Mayutan Arumaithurai and Xiaoming Fu, published at _SIGCOMM '17_][nfvnice_paper]. Note that this code does not contain the full Flurries or NFVnice systems, only the basic support for shared-CPU NFs.
+
+Usage:
+  - To enable shared core mode pass a `-c` flag to the onvm_mgr, and use a `-s` flag when starting a NF to specify that they can share cores with other NFs  
+
+Notes:
+  - All code for sharing CPUs is within `if (ONVM_NF_SHARE_CORE)` blocks
+  - When enabled, you can run multiple NFs on the same CPU core with much less interference than if they are polling for packets
+  - This code does not provide any particular intelligence for how NFs are scheduled or when they wakeup/sleep
+  - Note that the manager threads all still use polling
+
+### Major Architectural/API/Initialization/Signal Handling Changes:
```

Comment:
`Architectural Changes` should be one header, and `API changes` should be another. 

Under arch, you should expand high level on what has changed overall and how that changes the init sequence.  You should try your best to move all codeblocks and function protos under the API changes section.  IMO, arch changes section should give a very clear overview of what has changed and how, and why this change makes things better.

Under API, mention all of the new APIs and on a high-level note, why they make life better.  Also, signal handling changes should go here because it really hasn't changed, but the APIs to manage that component.